lib/core/client.js

```diff
@@ -159,7 +159,7 @@ class Client extends EventEmitter {
       }
     }
     this[kTLSSession] = null
-    this[kHostHeader] = `host: ${this[kUrl].hostname}\r\n`
+    this[kHostHeader] = `host: ${this[kUrl].host}\r\n`
```

Comment:
const host = this[kUrl].host ? this[kUrl].host : `${this[kUrl].hostname}${this[kUrl].port ? `:${this[kUrl].port}`}`

Reply:
Also probably an invalid arugment check in constructor to ensure that either host or hostname is set and a string